citron-collection-partitioner/collection_partitioner.py

```diff
@@ -0,0 +1,160 @@
+import pandas as pd
+import logging as log
+import re
+
+import boto3
+from botocore.exceptions import NoCredentialsError
+
+
+class CollectionPartitioner:
+    def __init__(self, bucket: str):
+        self.bucket = bucket
+
+    @staticmethod
+    def init_sts_client():
+        """Init client for AWS S3"""
+        try:
+            s3_client = boto3.client("sts")
+        except NoCredentialsError:
+            log.error("Credentials are missing or incorrect")
+            raise
+        return s3_client
+
+    def get_files_of_collection(self, collection_key: str) -> list:
+        """
+
+        :param collection_key: key of the collection folder
+        :return: list of keys of all the files in the collection folder
+        """
+
+        keys = []
+
+        sts_client= CollectionPartitioner.init_sts_client()
+        assumed_role_object = sts_client.assume_role(
+            RoleArn="arn:aws:iam::585762237892:role/gads-citron-modeo",
+            RoleSessionName="AssumeRoleSession1",
+        )
+        credentials = assumed_role_object["Credentials"]
+        s3_client = boto3.client(
+            "s3",
+            aws_access_key_id=credentials["AccessKeyId"],
+            aws_secret_access_key=credentials["SecretAccessKey"],
+            aws_session_token=credentials["SessionToken"],
+        )
+
+        for file in s3_client.list_objects(Bucket=self.bucket, Prefix= collection_key)["Contents"]:
+            if not file["Key"].endswith('/'):
+                keys.append(file["Key"])
+
+        sts_client.close()
+        s3_client.close()
+        return keys
+
+    @staticmethod
+    def get_file_infos(key: str) -> dict:
+        """
+
+        :param key: key of the file stored in s3
+        :return: dict containing the information of the file: key, year, month, day
+        """
+
+        res= re.search(r"^.*year=(?P<year>\d+)/month=(?P<month>\d+)/day=(?P<day>\d+)/(?P<file_name>.*)$",key)
+        file_infos={
+            "key": key,
+            "file_name":res.group('file_name'),
+            "year":int(res.group('year')),
+            "month":int(res.group('month')),
+            "day":int(res.group('day')),
+        }
+        return file_infos
+
+    def create_collection_structure(self, collection_key: str) -> pd.DataFrame:
+        """
+
+        :param collection_key: key of the collection in s3
+        :return: return a pandas dataframe that summarizes the information of each file in this collection
+        """
+        data= pd.DataFrame(columns=["key","file_name","year","month","day"])
+        files= self.get_files_of_collection(collection_key)
```

Comment:
Wrong spaces. You can use black to reformat your code : https://github.com/psf/black